src/main/java/org/intellij/sdk/codesync/HandleBuffer.java

```diff
@@ -0,0 +1,340 @@
+        }
+        return new DiffFile[0];
+    }
+
+    public static void scheduleBufferHandler() {
+        TimerTask task = new TimerTask() {
+            @Override
+            public void run() {
+                try {
+                    HandleBuffer.handleBuffer();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        };
+        Timer timer = new Timer(true);
+        timer.schedule(task, new Date(), 5000);
+    }
+
+    public static void handleBuffer() {
+        ConfigFile configFile;
+        HashSet<String> newFiles = new HashSet<>();
+
+        DiffFile[] diffFiles = getDiffFiles(DIFFS_REPO, ".yml");
+        if (diffFiles.length == 0) {
+            return;
+        }
+
+        CodeSyncClient client = new CodeSyncClient();
+        if (!client.isServerUp()) {
+            return;
+        }
+
+        diffFiles = Arrays.copyOfRange(
+            diffFiles, 0, diffFiles.length >= DIFFS_PER_ITERATION ? DIFFS_PER_ITERATION : diffFiles.length
+        );
+
+        try {
+            configFile = new ConfigFile(CONFIG_PATH);
+        } catch (InvalidConfigFileError error) {
+            System.out.printf("Config file error, %s%n", error.getMessage());
+            return;
+        }
+
+        for (DiffFile diffFile: diffFiles) {
+            if (!diffFile.isValid()) {
+                String filePath =  String.format("%s/%s", diffFile.repoPath, diffFile.fileRelativePath);
+                System.out.printf("Skipping invalid diff file: %s. data: %s", filePath, diffFile.diff);
+
+                diffFile.delete();
+                continue;
+            }
+
+            if (!configFile.repos.containsKey(diffFile.repoPath)) {
+                System.out.printf("Repo `%s` is in buffer.yml but not in configFile.yml", diffFile.repoPath);
+                continue;
+            }
+
+            ConfigRepo configRepo = configFile.getRepo(diffFile.repoPath);
+            if (!configRepo.branches.containsKey(diffFile.branch))  {
+                System.out.printf("Branch: `%s` is not synced for Repo `%s`", diffFile.branch, diffFile.repoPath);
+                continue;
+            }
+
+            if(diffFile.isDirRename) {
+                handleDirRename(diffFile);
+                diffFile.delete();
+                continue;
+            }
+
+            if (Utils.shouldIgnoreFile(diffFile.fileRelativePath, diffFile.repoPath)) {
+                diffFile.delete();
+                continue;
+            }
+
+            ConfigRepoBranch configRepoBranch = configRepo.getRepoBranch(diffFile.branch);
+            if (diffFile.isNewFile) {
+                newFiles.add(diffFile.fileRelativePath);
+                boolean isSuccess = handleNewFile(client, diffFile, configFile, configRepo, configRepoBranch);
+                if (isSuccess) {
+                    diffFile.delete();
+                }
```

Comment:
> 2- Add newly created file ID in the config file

this is happening inside `handleNewFile`

> 1- continue the loop

this is happening on the very next if condition.

Reply:
Looked at `handleNewFile`, as we read config file outside of for..loop, If next iteration of loop uses the updated config JSON then we are fine.
We just want to use updated config file for the next diff file so that it has the file ID of newly created file